File: leetCode/fukun/lc_112.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

var hasPathSum = function (root, sum) {
    const helper = function (node, add) {
        if (!node) return false;
        if (!node.left && !node.right && node.val + add === sum) return true;
        return helper(node.left, add + node.val) || helper(node.right, add + node.val);
    }
    return helper(root, 0);
};

let t = new TreeNode(1);
t.left = new TreeNode(2);
t.right = new TreeNode(3);
console.log(hasPathSum(t, 2));

File: leetCode/lwc/0101-0200/lc_112.java
package solution2;

/**
 * @author lwc
 * @date 2019/10/28 23:06
 */
public class U112 {
    public boolean hasPathSum(TreeNode root, int sum) {
        if(root == null) return false;
        if(root.val == sum && root.right == null && root.left == null)
            return true;
        return hasPathSum(root.left, sum - root.val) || hasPathSum(root.right, sum - root.val);
    }
}
